### PR TITLE
Convert Name Resolution document to markdown

### PR DESCRIPTION
**Pull request by laurentlb**
I've also added a Backward-compatibility section in the document.

### CODE REVIEW

File: docs/2018-06-18-name-resolution.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
+# Name resolution in Skylark
+
+> Author: [laurentlb](https://github.com/laurentlb/)<br/>
```
Comment (jin) [inline docs/2018-06-18-name-resolution.md]: Do you think it's good to add the header metadata like what we do in the blogposts, e.g. https://github.com/bazelbuild/bazel-blog/blob/master/_posts/2015-04-22-thank-you-stickers.md? 

```
---
created: 2018-06-18
status: Approved
reviewers:
  - dslomov
  - alandonovan
title: Name resolution in Skylark
authors:
  - laurentlb
---
```

GitHub renders it specially, and we can easily parse it programmatically in the future.

Comment (laurentlb) [reply docs/2018-06-18-name-resolution.md]: done

**Comment by laurentlb** [comment]
Please take another look

File: docs/2018-06-18-name-resolution.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+This document lists the differences between Bazel implementation of Skylark and
+the [language specification](https://docs.bazel.build/versions/master/skylark/spec.html),
+in relation to name resolution, shadowing, and rebinding.
+
+
+## Shadowing of built-ins
+
+The specification mentions there are 4 blocks:
+
+*   Universe block (the built-ins)
```
Comment (philwo) [inline docs/2018-06-18-name-resolution.md]: Why is the indentation of lists here different than below?

File: docs/2018-06-18-name-resolution.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+**Resolution**: Update Bazel, forbid rebinding of global variables in BUILD files.
+
+This is a **breaking change** for Bazel.
+
+## Backward-compatibility
+
+Two sections in this document constitute breaking changes to Bazel:
+
+ * [Lexical binding of variables](#lexical-binding-of-variables) is fixing a bug
```
Comment (philwo) [inline docs/2018-06-18-name-resolution.md]: (This is the other list that I'm referring to above.)

File: docs/2018-06-18-name-resolution.md (hunk 2 above)
Comment (laurentlb) [reply docs/2018-06-18-name-resolution.md]: fixed

**Pull request merged**